lib/tasks/rescan_reports_origins.rake

```diff
@@ -0,0 +1,23 @@
+desc 'Rescan existing reports without origin and tries to find correct origin'
+task :rescan_reports_origins do
+  puts "Scanning #{ConfigReport.count} reports, this can take a long time, it's safe to interrupt and rerun later..."
+  User.as_anonymous_admin do
+    ConfigReport.where(:origin => nil).includes(:logs => [:source, :message]).find_in_batches(batch_size: 100) do |group|
+      group.each do |report|
+        Foreman::Plugin.report_scanner_registry.report_scanners.each do |scanner|
+          logs = report.logs.map do |log|
```

Comment:
On second look, should this maybe be limited to only the last log?

Reply:
Nah, we need all. :D `PuppetReportScanner` only needs the first, but the AnsibleReportScanner required more to scan.

Reply:
:dart: :-)